### ISSUE TITLE
cf push stops working with Eirini after a while (was stats server unavailable)

### ISSUE THREAD

**Issue by gaktive** [cloudfoundry-incubator/kubecf#1547]
**Describe the bug**
Placeholder, per @satadruroy's findings -- we've received a couple reports that when deploying KubeCF with Eirini, the Stratos Metrics will not be able to get consistent results. It points to the status API not working after a bit of time.

**To Reproduce**
- Deploy KubeCF with Eirini
- Deploy Stratos Metrics onto it

Workaround: restart the api pod

**Comment by satadruroy** [comment]
Reported from the field:

"intermittent failures in doing a `cf push` with eirini and also losing stats in Stratos:

```
{
  "error": {
    "statusCode": 503,
    "status": "503 Service Unavailable"
  },
  "errorResponse": {
    "description": "Stats unavailable: Stats server temporarily unavailable.",
    "error_code": "CF-StatsUnavailable",
    "code": 200002
  }
}
```

A restart of the `api-0` pod seems to resolve the issue but it re-appears. The following log messages are seen in the `api-0` pod logs when the error happens:

```
logs 2020/10/30 15:38:28, /var/vcap/sys/log/cloud_controller_ng/nginx-access.log, api.cap.suse.ninja - [30/Oct/2020:15:38:28 +0000] "GET /v2/buildpacks HTTP/1.1" 200 5206 "-" "cf_exporter/0.10.0" 10.244.2.193, 10.244.2.152, 10.244.3.149 vcap_request_id:84c558c3-7971-4dc7-54ed-d2ab4d05821d::ea3f8365-7e1f-43bc-b67f-876f898afa29 response_time:0.652
cc-uploader-cc-uploader 2020/10/30 09:33:15 http: TLS handshake error from 10.244.4.146:60090: EOF
cc-uploader-cc-uploader 2020/10/30 09:33:25 http: TLS handshake error from 10.244.4.146:60154: EOF
cc-uploader-cc-uploader 2020/10/30 09:33:35 http: TLS handshake error from 10.244.4.146:60300: EOF
file-server-file-server {"timestamp":"2020-10-30T11:08:09.767831243Z","level":"info","source":"file-server","message":"file-server.static-file.response","data":{"method":"GET","session":"7039","size":174369,"status":200,"uri":"/v1/static/file_server/bin/file-server"}}
cc-uploader-cc-uploader 2020/10/30 09:33:45 http: TLS handshake error from 10.244.4.146:60382: EOF
route-registrar-route-registrar {"timestamp":"2020-10-30T15:32:22.373843280Z","level":"info","source":"Route Registrar","message":"Route Registrar.Registering route","data":{"route":{"Type":"","Name":"policy-server","Port":null,"TLSPort":4002,"Tags":null,"URIs":["api.cap.suse.ninja/networking"],"RouterGroup":"","Host":"","ExternalPort":null,"RouteServiceUrl":"","RegistrationInterval":20000000000,"HealthCheck":null,"ServerCertDomainSAN":"api.cap.suse.ninja"}}}
```
Not sure if the TLS handshake error is a genuine issue or a red herring.

**Comment by satadruroy** [comment]
I was finally able to repro this problem on both AKS and EKS so this is not unique to CaaSP. My setup:

AKS/EKS, Eirini, SA.

Deploy app - works ok. Leave app running overnight.

Check `cf app` - stats server is unavailable. `cf push` also fails with the error `Stats unavailable: Stats server temporarily unavailable.` 

I noticed the log-cache pod is now 9/10 with the `log-cache-log-cache` container reporting not ready. Restarting the pod does not resolve the problem. Restarting the `api-0` pod resolves the issue.

**Comment by andreas-kupries** [comment]
Note that `TLS handshake error` are generally benign. They are the result of various live/readiness probes. The probes are done by opening a connection to the port, and then closing it again. This verifies that the port is reachable and responding. On the server side however this is a TLS handshake violation, as it does not complete the handshake at all, and reported as such.

**Comment by jandubois** [comment]
@mook-as and I have confirmed that the error condition is triggered when `log-cache` is not ready (by scaling it down to 0 instances) and the CC tries to access app metrics, e.g. by running `cf apps`, or `cf restart`, or `cf push`.

The error persists even when `log-cache` is scaled back to 1 instance, and the containers are all ready. It is necessary to restart the `cloud_controller_ng` container in the `api` pod to restore access to the metrics functionality.

**Comment by jandubois** [comment]
I've now confirmed the same behaviour with Diego (on minikube): scale down `log-cache`, run `cf apps` to trigger the error, scale `log-cache` back up, and metrics remain broken, `cf app XXX`, `cf apps`, `cf push` etc all no longer work.

**Comment by mook-as** [comment]
Should be fixed by #1650 — please reopen if you still see this.

Please note that it can take a while for things to stabilize; a backoff timer is involved, so the longer the `log-cache` outage was, the longer it will stay broken.  But since in this case it was offline for an unknown period of time, by the time we checked again it should already be fixed.
